File: Practice/atcoder/AGC/002/src/b.py
# 赤が入ってる可能性が0になる場合（すべての球を取り出した場合）があることに注意
n, m = map(int, input().split())
balls = [1 for _ in range(n)]
reddable = [False for _ in range(n)]
reddable[0] = True
for _ in range(m):
    x, y = map(lambda x: int(x) - 1, input().split())
    if reddable[x] == True:
        reddable[y] = True
        balls[x] -= 1
        balls[y] += 1
    else:
        balls[x] -= 1
        balls[y] += 1
    if balls[x] == 0:
        reddable[x] = False

ans = list(filter(lambda x: x == True, reddable))
print(len(ans))

File: Practice/atcoder/AGC/002/src/b.go
// AGCは質が高くていいな〜〜
// 枯渇しない限りは拡散できる
// 最初1-indexedにしてたのに0に赤を置いてて無限時間バグった

package main

import "fmt"

func main() {
	var N, M int
	fmt.Scan(&N, &M)
	Balls := make([]int, N+1)
	PossibilityRed := make([]bool, N+1)
	for i := 1; i <= N; i++ {
		Balls[i] = 1
		PossibilityRed[i] = false // いらんけど
	}
	PossibilityRed[1] = true
	for i := 0; i < M; i++ {
		var A, B int
		fmt.Scan(&A, &B)
		if Balls[A] == 0 {
			continue // あるかどうかしらんけど
		}
		if PossibilityRed[A] {
			PossibilityRed[B] = true
			if Balls[A] == 1 {
				PossibilityRed[A] = false
			}
		}
		Balls[A]--
		Balls[B]++
	}
	ans := 0
	for _, p := range PossibilityRed {
		if p {
			ans++
		}
	}
	fmt.Println(ans)
}
